'Security' section with all topics (#416)
hyperledger/iroha-2-docs: src/guide/security/storing-cryptographic-keys.md

### Using SSH and SSH Agent

**Secure Shell Protocol** (`SSH`) is a cryptographic network protocol that serves as a virtual gateway, enabling secure access to remote machines via potentially not-so-secure networks by using SSH keys—access credentials. It provides an efficient way to remotely interact with systems without the necessity of physical presence. In this context, `SSH` offers two primary authentication mechanisms: the conventional password-based approach and the more secure public-private key pair method.

For more information on `SSH`, see [the related SSH Academy topic](https://www.ssh.com/academy/ssh).

To streamline the login process and bypass the need for repetitive input, it is possible to pair the `SSH` keys with the **SSH Agent** (`ssh-agent`)—the assistant program that remembers your `SSH` keys and/or password for the duration of a session. This setup permits the `SSH` gateway to effortlessly access the keys whenever it connects to other machines.

> The workflow here is as follows: you have your public key stored on a remote system, and you keep your private key secure in your possession. Whenever you want to access a remote system, the `ssh-agent` steps in to communicate your _public_ key to the accessed system. The remote system then sends back a [challenge](https://en.wikipedia.org/wiki/Challenge%E2%80%93response_authentication) that only your _private_ key can properly respond to. Your `ssh-agent` handles this challenge by using your _private_ key and sends the correct response back to the remote system. If the response matches what the system expected, you're granted access.

Reviewer: ```suggestion
The workflow here is as follows: you have your public key stored on a remote system and keep your private key secure. Whenever you want to access a remote system, the `ssh-agent` steps in to communicate your _public_ key to the accessed system. The remote system then sends back a [challenge](https://en.wikipedia.org/wiki/Challenge%E2%80%93response_authentication) that only your _private_ key can properly respond to. Your `ssh-agent` handles this challenge by using your _private_ key and sends the correct response back to the remote system. If the response matches what the system expected, you're granted access.
```

Reviewer: Maybe this could also be formatted as bullets or a numbered list instead of a paragraph.

The beauty of the `ssh-agent` is that it holds onto your private key during your session, so there is no need to keep entering your password or private key passphrase every time you connect to a remote system.

> For more information on `ssh-agent`, see [the related SSH Academy topic](https://www.ssh.com/academy/ssh/agent).

Reviewer: ```suggestion
For more information on the `ssh-agent`, see [the related SSH Academy topic](https://www.ssh.com/academy/ssh/agent).
```

::: info Note

For a detailed overview of the `SSH` protocol and the `ssh-agent` tool, see the following [SSH Academy](https://www.ssh.com/academy) topics:

  - [What is SSH (Secure Shell)?](https://www.ssh.com/academy/ssh)
  - [ssh-agent: How to configure ssh-agent, agent forwarding, & agent protocol](https://www.ssh.com/academy/ssh/agent)

:::

### Adding a Password Manager Program

It is recommended to enhance the security of your `SSH` keys by protecting them with a password, which acts as an additional obstacle in the way of malicious parties aiming to obtain your sensitive information.

> A variety of password managers can be used for the temporary storage of the user passwords and `SSH` keys. For the sake of clarity, [KeePass](https://keepass.info/) is used as an example password manager, specifically its [KeePassXC](https://keepassxc.org/) port running on Linux-based operating systems.

Reviewer: ```suggestion
A variety of password managers can be used to store user passwords and `SSH` keys temporarily. For the sake of clarity, [KeePass](https://keepass.info/) is used as an example password manager, specifically, the [KeePassXC](https://keepassxc.org/) port running on Linux-based operating systems.
```

4. For every key that you would like to store in the KeePassXC Database you created, perform the following steps:

   - Add a new entry in the database. For instructions, see [KeePassXC User Guide > Creating Your First Database](https://keepassxc.org/docs/KeePassXC_UserGuide#_creating_your_first_database).

   - When adding a new entry, attach the file containing the key by doing the following: select **Advanced** from the left menu, then select **Add** in the **Attachments** section, choose the required file in the **Select files** window that appears.

   - When adding a new entry, select **SSH Agent** from the left menu, then select the key file you added from the **Attachment** menu in the **Private key** section; then select the following checkboxes:

      - **Add key to agent when database is opened/unlocked**

      - **Remove key from agent when database is closed/locked**

      - **Require user confirmation when this key is used**

   - If necessary, make other changes to the entry.

   - When ready, select **OK** to save the entry.

>    ::: details Show reference screenshots

Reviewer: ```suggestion
   ::: details Here are reference screenshots
```

Reviewer 2: This is an expandable section, "show" is a better option.
